sympy/crypto/crypto.py

```diff
@@ -3324,3 +3324,86 @@ def decipher_bg(message, key):
         orig_msg += (m ^ b)
 
     return orig_msg
+
+
+########### XOR Cipher ############
+
+def encipher_xor(message, key):
+    """
+    Performs XOR encryption on plaintext and returns ciphertext.
+
+    Examples
+    ========
+
+    >>> from sympy.crypto.crypto import encipher_xor
+    >>> encipher_xor("THISISASTRING", "ABCD")
+    '150a0a170811021715100a0a06'
+
+    Parameters
+    ==========
+
+    message: str
+        the message to be encrypted.
+    key: str
+        the key to be used for encryption of the message.
+
+    Returns
+    =======
+    The encrypted ciphertext.
+
+    References
+    ==========
+    .. [1] https://en.wikipedia.org/wiki/XOR_cipher
+
+    """
+
+    encrypted_text = ""
+    itr = 0
+    for f in range(len(message)):
+        encrypted_text += hex(ord(key[itr]) ^ ord(message[f]))[2:].zfill(2)
```

Comment:
```suggestion
        encrypted_text += hex(ord(key[f%len(key)]) ^ ord(message[f]))[2:].zfill(2)
```
And then the handling of itr is not needed.